# LEAGUE DATA SET 에서 아이언 2티어 유저데이터 받아오기

In [1]:
import requests
import json
import pandas as pd
import numpy as np
import pymongo

# mongodb에서의 db와 collection 변수에 저장

In [6]:
from pymongo import MongoClient
client = MongoClient()
db = client.RiotAPI
collection = db.league_iron2_user_test

# RiotAPI에서 아이언 2티어 유저데이터 받아오기

In [3]:
api_key = 'RGAPI-f2e2dae1-7bc9-4372-8181-85129b79b961'
Tier_Iron = "https://kr.api.riotgames.com/lol/league/v4/entries/RANKED_SOLO_5x5/IRON/II?page=1&api_key="+api_key
r = requests.get(Tier_Iron)
r.json()

[{'leagueId': '7b428990-30a5-4868-a240-2345482a6aef',
  'queueType': 'RANKED_SOLO_5x5',
  'tier': 'IRON',
  'rank': 'II',
  'summonerId': 'DeTnPQuV6Y2Mno1paCEMw3ZQG_JMNmo7qgL_EJ7MYlT24gqC',
  'summonerName': 'SKT T2 Sorry',
  'leaguePoints': 46,
  'wins': 7,
  'losses': 10,
  'veteran': False,
  'inactive': False,
  'freshBlood': False,
  'hotStreak': False},
 {'leagueId': '7bd0802f-28af-408a-841f-e82110197206',
  'queueType': 'RANKED_SOLO_5x5',
  'tier': 'IRON',
  'rank': 'II',
  'summonerId': '_XR733VqZn7AZP1HeZOPNT4FA5Wk9sgNK8zSzCnhXqlmkbn0',
  'summonerName': 'NEXON EA SPORTS',
  'leaguePoints': 76,
  'wins': 16,
  'losses': 28,
  'veteran': False,
  'inactive': False,
  'freshBlood': False,
  'hotStreak': False},
 {'leagueId': '01407cdc-5128-4cb2-8fa3-637dfab2d120',
  'queueType': 'RANKED_SOLO_5x5',
  'tier': 'IRON',
  'rank': 'II',
  'summonerId': '94qoDLUjzMuoM7mC41WuObpaXuRqpDkJDy-6lpyPQ3LjnoaW',
  'summonerName': '딜나와',
  'leaguePoints': 7,
  'wins': 209,
  'losses': 224,
  'v

# json형식의 데이터를 데이터프레임으로 저장

In [4]:
iron_league_df = pd.DataFrame(r.json())
iron_league_df

leagueId        queueType  tier rank  \
0    7b428990-30a5-4868-a240-2345482a6aef  RANKED_SOLO_5x5  IRON   II   
1    7bd0802f-28af-408a-841f-e82110197206  RANKED_SOLO_5x5  IRON   II   
2    01407cdc-5128-4cb2-8fa3-637dfab2d120  RANKED_SOLO_5x5  IRON   II   
3    ea9d7413-be94-4954-a6b3-86d7c993de84  RANKED_SOLO_5x5  IRON   II   
4    bb627f5b-c64b-4bdc-beaa-c01f2bc0a4b7  RANKED_SOLO_5x5  IRON   II   
..                                    ...              ...   ...  ...   
200  16f97cda-445e-4c5f-9d5c-9733c388b2dc  RANKED_SOLO_5x5  IRON   II   
201  b4b2ba47-ddbf-46d4-b9b2-152135b1301c  RANKED_SOLO_5x5  IRON   II   
202  df3f98f2-83db-49f4-8ad6-2cacfcb5d5f6  RANKED_SOLO_5x5  IRON   II   
203  86fbbc6b-eb0c-4075-a2dc-3c1b8a7ae0f6  RANKED_SOLO_5x5  IRON   II   
204  17151d31-34d9-4262-ab29-0ef02752365f  RANKED_SOLO_5x5  IRON   II   

                                           summonerId     summonerName  \
0    DeTnPQuV6Y2Mno1paCEMw3ZQG_JMNmo7qgL_EJ7MYlT24gqC     SKT T2 Sorry   
1    _XR733VqZn7AZP1HeZOPNT4FA5Wk9sgNK8zSzCnhXqlmkbn0  NEXON EA SPORTS   
2    94qoDLUjzMuoM7mC41WuObpaXuRqpDkJDy-6lpyPQ3LjnoaW              딜나와   
3    wp6aTiTg3VClJjYJ2HmjyuBfkJrmFcZVoQmR3WvJCN_6RwK4          빵뎅이흔드록바   
4    Ary-2PyAMocDgNh-0jfMBLvBnkuB2HUNLTlo5gnefWcf7NeZ          tamtoru   
..                                                ...              ...   
200  MTU3tjOTsEOTyM_5SOIC1xWxqRQPn-5vsNmxxCgdLnV24zo4          나는야 민준이   
201  -7hMM2fGD93pnzjgjbjWsQodous0RISfSZYujV9aqVGMYI92           fgrr07   
202  6aQGB6z6bFX5TuBFv8B_u1dEvEiHWBhrbd8iPf16QrzUPd7P           화이트라이온   
203  ExDJe9-wR5N7xeXUh_08YEAtJs57D_jrIQfKFuPjAWkMvjQp           너오늘똥밟음   
204  JZQXihNL5Qn4Zb43V77uExKF1F2WYKd7iDmh4u14G3PhFxV_          움직여라두뇌야   

     leaguePoints  wins  losses  veteran  inactive  freshBlood  hotStreak  
0              46     7      10    False     False       False      False  
1              76    16      28    False     False       False      False  
2               7   209     224    False     False       False      False  
3              54     1      10    False     False       False      False  
4              77     5      13    False     False       False      False  
..            ...   ...     ...      ...       ...         ...        ...  
200            48    16      31    False     False       False      False  
201            76    63      80    False     False       False      False  
202            43    55      61    False     False       False      False  
203            60    47      56    False     False       False      False  
204            38     3      17    False     False       False      False  

[205 rows x 13 columns]

# dataframe을 딕셔너리형태로 변환해서 collection에 저장

In [7]:
collection.insert_many(iron_league_df.to_dict("records"))
collection

Collection(Database(MongoClient(host=['localhost:27017'], document_class=dict, tz_aware=False, connect=True), 'RiotAPI'), 'league_iron2_user_test')

In [8]:
for i in collection.find():
    print(i)

{'_id': ObjectId('606d57af9e63488f51772138'), 'leagueId': '7b428990-30a5-4868-a240-2345482a6aef', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'IRON', 'rank': 'II', 'summonerId': 'DeTnPQuV6Y2Mno1paCEMw3ZQG_JMNmo7qgL_EJ7MYlT24gqC', 'summonerName': 'SKT T2 Sorry', 'leaguePoints': 46, 'wins': 7, 'losses': 10, 'veteran': False, 'inactive': False, 'freshBlood': False, 'hotStreak': False}
{'_id': ObjectId('606d57af9e63488f51772139'), 'leagueId': '7bd0802f-28af-408a-841f-e82110197206', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'IRON', 'rank': 'II', 'summonerId': '_XR733VqZn7AZP1HeZOPNT4FA5Wk9sgNK8zSzCnhXqlmkbn0', 'summonerName': 'NEXON EA SPORTS', 'leaguePoints': 76, 'wins': 16, 'losses': 28, 'veteran': False, 'inactive': False, 'freshBlood': False, 'hotStreak': False}
{'_id': ObjectId('606d57af9e63488f5177213a'), 'leagueId': '01407cdc-5128-4cb2-8fa3-637dfab2d120', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'IRON', 'rank': 'II', 'summonerId': '94qoDLUjzMuoM7mC41WuObpaXuRqpDkJDy-6lpyPQ3LjnoaW', 'summone

# collection 데이터수정
## test db의 monsters collection 데이터 수정

In [9]:
db_test = client.test
collection_monsters = db_test.monsters
for i in collection_monsters.find():
    print(i)

{'_id': ObjectId('6049cbb23fc78dc3eaa075e8'), 'name': 'Slime', 'hp': 25.0, 'xp': 10.0, 'att': 10.0}
{'_id': ObjectId('6049cbb93fc78dc3eaa075e9'), 'name': 'Skeleton', 'hp': 50.0, 'xp': 20.0, 'att': 20.0}
{'_id': ObjectId('6049cbb93fc78dc3eaa075ea'), 'name': 'Demon', 'hp': 500.0, 'xp': 200.0, 'att': 200.0}
{'_id': ObjectId('606d55814e8b648c6e9f0f88'), 'leagueId': '7b428990-30a5-4868-a240-2345482a6aef', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'IRON', 'rank': 'II', 'summonerId': 'DeTnPQuV6Y2Mno1paCEMw3ZQG_JMNmo7qgL_EJ7MYlT24gqC', 'summonerName': 'SKT T2 Sorry', 'leaguePoints': 46, 'wins': 7, 'losses': 10, 'veteran': False, 'inactive': False, 'freshBlood': False, 'hotStreak': False}
{'_id': ObjectId('606d55814e8b648c6e9f0f89'), 'leagueId': '7bd0802f-28af-408a-841f-e82110197206', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'IRON', 'rank': 'II', 'summonerId': '_XR733VqZn7AZP1HeZOPNT4FA5Wk9sgNK8zSzCnhXqlmkbn0', 'summonerName': 'NEXON EA SPORTS', 'leaguePoints': 76, 'wins': 16, 'losses': 28, 'veter

In [12]:
collection_monsters.find()[1]

{'_id': ObjectId('6049cbb93fc78dc3eaa075e9'),
 'name': 'Skeleton',
 'hp': 50.0,
 'xp': 20.0,
 'att': 20.0}

# mongodb collections ㅡ> dataframe
* mongodb의 monstrers collection을 리스트형태로 바꾼 후 데이터프레임으로 저장

In [14]:
test_df = pd.DataFrame(list(collection_monsters.find()))
test_df

_id      name     hp     xp    att  \
0    6049cbb23fc78dc3eaa075e8     Slime   25.0   10.0   10.0   
1    6049cbb93fc78dc3eaa075e9  Skeleton   50.0   20.0   20.0   
2    6049cbb93fc78dc3eaa075ea     Demon  500.0  200.0  200.0   
3    606d55814e8b648c6e9f0f88       NaN    NaN    NaN    NaN   
4    606d55814e8b648c6e9f0f89       NaN    NaN    NaN    NaN   
..                        ...       ...    ...    ...    ...   
203  606d55814e8b648c6e9f1050       NaN    NaN    NaN    NaN   
204  606d55814e8b648c6e9f1051       NaN    NaN    NaN    NaN   
205  606d55814e8b648c6e9f1052       NaN    NaN    NaN    NaN   
206  606d55814e8b648c6e9f1053       NaN    NaN    NaN    NaN   
207  606d55814e8b648c6e9f1054       NaN    NaN    NaN    NaN   

                                 leagueId        queueType  tier rank  \
0                                     NaN              NaN   NaN  NaN   
1                                     NaN              NaN   NaN  NaN   
2                                     NaN              NaN   NaN  NaN   
3    7b428990-30a5-4868-a240-2345482a6aef  RANKED_SOLO_5x5  IRON   II   
4    7bd0802f-28af-408a-841f-e82110197206  RANKED_SOLO_5x5  IRON   II   
..                                    ...              ...   ...  ...   
203  b4b2ba47-ddbf-46d4-b9b2-152135b1301c  RANKED_SOLO_5x5  IRON   II   
204  df3f98f2-83db-49f4-8ad6-2cacfcb5d5f6  RANKED_SOLO_5x5  IRON   II   
205  86fbbc6b-eb0c-4075-a2dc-3c1b8a7ae0f6  RANKED_SOLO_5x5  IRON   II   
206  17151d31-34d9-4262-ab29-0ef02752365f  RANKED_SOLO_5x5  IRON   II   
207  fdb38307-f7e1-4766-9e34-594e98446109  RANKED_SOLO_5x5  IRON   II   

                                           summonerId     summonerName  \
0                                                 NaN              NaN   
1                                                 NaN              NaN   
2                                                 NaN              NaN   
3    DeTnPQuV6Y2Mno1paCEMw3ZQG_JMNmo7qgL_EJ7MYlT24gqC     SKT T2 Sorry   
4    _XR733VqZn7AZP1HeZOPNT4FA5Wk9sgNK8zSzCnhXqlmkbn0  NEXON EA SPORTS   
..                                                ...              ...   
203  -7hMM2fGD93pnzjgjbjWsQodous0RISfSZYujV9aqVGMYI92           fgrr07   
204  6aQGB6z6bFX5TuBFv8B_u1dEvEiHWBhrbd8iPf16QrzUPd7P           화이트라이온   
205  ExDJe9-wR5N7xeXUh_08YEAtJs57D_jrIQfKFuPjAWkMvjQp           너오늘똥밟음   
206  JZQXihNL5Qn4Zb43V77uExKF1F2WYKd7iDmh4u14G3PhFxV_          움직여라두뇌야   
207  W3VV5Lbqlb1oz6dB4AREeLTVXvAiVrbRnyIGatQB7ZSGtSrH           96대장이다   

     leaguePoints  wins  losses veteran inactive freshBlood hotStreak  
0             NaN   NaN     NaN     NaN      NaN        NaN       NaN  
1             NaN   NaN     NaN     NaN      NaN        NaN       NaN  
2             NaN   NaN     NaN     NaN      NaN        NaN       NaN  
3            46.0   7.0    10.0   False    False      False     False  
4            76.0  16.0    28.0   False    False      False     False  
..            ...   ...     ...     ...      ...        ...       ...  
203          76.0  63.0    80.0   False    False      False     False  
204          43.0  55.0    61.0   False    False      False     False  
205          60.0  47.0    56.0   False    False      False     False  
206          38.0   3.0    17.0   False    False      False     False  
207          68.0   4.0     6.0   False    False      False     False  

[208 rows x 18 columns]

# dataframe 전처리
## dataframe 행 삭제
* for문을 이용하여 index가 3부터207까지 삭제

In [24]:
df = test_df
for i in range(3,208):
    df = df.drop(index=i,axis=0)
df

_id      name     hp     xp    att leagueId queueType  \
0  6049cbb23fc78dc3eaa075e8     Slime   25.0   10.0   10.0      NaN       NaN   
1  6049cbb93fc78dc3eaa075e9  Skeleton   50.0   20.0   20.0      NaN       NaN   
2  6049cbb93fc78dc3eaa075ea     Demon  500.0  200.0  200.0      NaN       NaN   

  tier rank summonerId summonerName  leaguePoints  wins  losses veteran  \
0  NaN  NaN        NaN          NaN           NaN   NaN     NaN     NaN   
1  NaN  NaN        NaN          NaN           NaN   NaN     NaN     NaN   
2  NaN  NaN        NaN          NaN           NaN   NaN     NaN     NaN   

  inactive freshBlood hotStreak  
0      NaN        NaN       NaN  
1      NaN        NaN       NaN  
2      NaN        NaN       NaN

## dataframe 열 삭제
* 컬럼을 리스트에 담아서 5번째 컬럼부터 모두삭제

In [30]:
df_columns_list=list(df.columns)
df_columns_list

['_id',
 'name',
 'hp',
 'xp',
 'att',
 'rank',
 'summonerId',
 'summonerName',
 'leaguePoints',
 'wins',
 'losses',
 'veteran',
 'inactive',
 'freshBlood',
 'hotStreak']

In [31]:
df = df.drop(columns=df_columns_list[5:],axis=1)
df

_id      name     hp     xp    att
0  6049cbb23fc78dc3eaa075e8     Slime   25.0   10.0   10.0
1  6049cbb93fc78dc3eaa075e9  Skeleton   50.0   20.0   20.0
2  6049cbb93fc78dc3eaa075ea     Demon  500.0  200.0  200.0

# 컬렉션 삭제

In [33]:
collection_monsters.drop()

# 전처리된 dataframe을 mongodb에 삽입
* dataframe을 딕셔너리 형태로 바꿔준다

In [35]:
collection_monsters.insert_many(df.to_dict("records"))

In [37]:
for i in collection_monsters.find():
    print(i)

{'_id': ObjectId('6049cbb23fc78dc3eaa075e8'), 'name': 'Slime', 'hp': 25.0, 'xp': 10.0, 'att': 10.0}
{'_id': ObjectId('6049cbb93fc78dc3eaa075e9'), 'name': 'Skeleton', 'hp': 50.0, 'xp': 20.0, 'att': 20.0}
{'_id': ObjectId('6049cbb93fc78dc3eaa075ea'), 'name': 'Demon', 'hp': 500.0, 'xp': 200.0, 'att': 200.0}
